packages/concerto-core/lib/decoratormanager.js

```diff
@@ -200,26 +201,80 @@ class DecoratorManager {
     }
 
     /**
-     * Applies all the decorator commands from the DecoratorCommandSet
-     * to the ModelManager.
+     * Add decorator commands set with index object to the coresponding target map
+     * @param {*} targetMap the target map to add the command to
+     * @param {targetKey} targetKey the target key to add the command to
+     * @param {DcsIndexWrapper} dcsWithIndex the command to add
+     * @private
+     */
+    static addDcsWithIndexToMap(targetMap, targetKey, dcsWithIndex) {
+        const targetCommands = targetMap.get(targetKey);
+        if (targetCommands) {
+            targetCommands.push(dcsWithIndex);
+        } else {
+            targetMap.set(targetKey, [dcsWithIndex]);
+        }
+    }
+
+
+    /**
+     * Creates five difeerent maps to index decorator command sets by target type and returns them
+     * @param {*} decoratorCommandSet the DecoratorCommandSet object
+     * @returns {Object} a new model manager with the decorations applied
+     * @private
+     */
+    static getDecoratorMaps(decoratorCommandSet) {
+        const namespaceCommandsMap = new Map();
+        const declarationCommandsMap = new Map();
+        const propertyCommandsMap = new Map();
+        const mapElementCommandsMap = new Map();
+        const typeCommandsMap = new Map();
+
+        decoratorCommandSet.commands.map((decoratorCommand, index) => {
+            const dcsWithIndex = new DcsIndexWrapper(decoratorCommand, index);
+            if(decoratorCommand.target.type) {
```

Comment:
Is there an advantage to using a switch statement here?

Reply:
Also, how do we protect against falsy, but non-null values in these conditions?

Reply:
> Is there an advantage to using a switch statement here?

Yes, refactored it to have switch case.

> Also, how do we protect against falsy, but non-null values in these conditions?

They won't get added to their respective maps. `target.[elementTarget]` always needs to be string to be valid, if non-null falsy, it won't be added to any of the maps.